lisp/mastodon-toot.el

```diff
@@ -142,14 +141,61 @@ Set `mastodon-toot--content-warning' to nil."
       (mastodon-http--triage response
                              (lambda () (message "Toot toot!"))))))
 
+(defun mastodon-toot--get-acct ()
+  "Adds the acct for the handle under the current point to the Kill ring.
+
+This only works if the html received is formatted to contain the domain. 
+it wont work for copying accts from the by-line or if someone entered a 
+handle manually"
+  (interactive)
+  (let ((user-address (plist-get (text-properties-at (point)) 'shr-url)))
+    (if (stringp user-address)
+        (let* ((split-address (split-string-and-unquote user-address "/"))
+               (domain (elt split-address 1))
+               (username (string-remove-prefix "@" (car (last split-address))))
+               (acct (concat "@" username "@" domain)))
+          (kill-new acct)
+          (message (format "Added %s to the kill ring." acct))
+          t)
+      (progn (message (format "No properly formated acct at point:%s." (point)))
+             nil))))
+
+(defun mastodon-toot--process-local (acct)
+  "Adds domain to local ACCT and replaces the curent user name with \"\".
+
+Mastodon requires the full user@domain, even in the case of local accts. 
+eg. \"user\" -> \"user@local.social \" (when local.social is the domain of the 
+mastodon-instance-url).
+eg. \"yourusername\" -> \"\"
+eg. \"feduser@fed.social\" -> \"feduser@fed.social\""
+  (cond ((string-match-p "@" acct) (concat "@" acct " ")) ; federated acct
+        ((string-match-p (mastodon-auth--user-acct) acct) "") ; your acct
+        (t (concat "@" acct "@" ; local acct
+                   (cadr (split-string mastodon-instance-url "/" t)) " "))))
+
+(defun mastodon-toot--mentions (status)
+  "Extract mentions from STATUS and process them into a string."
+  (interactive)
+  (let ((mentions (cdr (assoc 'mentions status))))
+    (if (> (length mentions) 0)
```

Comment:
Nit: This check isn't needed, right?  If mentions is the empty vector, the mapappend will still do the right thing.